src/plugins/userIndicator/components/UserIndicator.vue

```diff
@@ -65,18 +65,28 @@ export default {
       const UserAPI = this.openmct.user;
       const activeRole = UserAPI.getActiveRole();
       this.role = activeRole;
-      if (!activeRole) {
-        this.promptForRoleSelection();
+      this.availableRoles = await this.openmct.user.getPossibleRoles();
+
+      // clear role if it's not in list of available roles, e.g., removed by admin
+      if (!this.availableRoles?.includes(this.role)) {
+        this.role = null;
+        UserAPI.setActiveRole(null);
```

Comment:
Local storage will coalesce null into a string i.e. "null". So we might want to update the UserAPI to clear the local storage value instead.
```
  setActiveRole(role) {
    if (!role) {
      StoragePersistance.clearActiveRole();
    } else {
      StoragePersistance.setActiveRole(role);
    }
    this.emit('roleChanged', role);
  }
```

Reply:
nice catch!